Home_Work_15/HW15.cpp

```diff
@@ -29,128 +27,123 @@ int sumStudentsPassedTreshold(Student arrayOfStudents[amountOfStudents], const i
+	std::cout << "Avarage rate is: " << studentAverageRates << std::endl << std::endl;
 
-    Student arrayOfStudents[amountOfStudents] = { {"Dwight",{4,5,3,5}} , {"Meg", {5,5,5,5}}, {"Bill", {5,5,5,5}}, {"Steve", {5,3,3,2}}, {"Nancy", {1,3,4,5}} };
+	Student arrayOfStudents[amountOfStudents] = { {"Dwight",{4,5,3,5}} , {"Meg", {5,5,5,5}}, {"Bill", {5,5,5,5}}, {"Steve", {5,3,3,2}}, {"Nancy", {1,3,4,5}} };
 
-    std::sort(std::begin(arrayOfStudents), std::end(arrayOfStudents), sortByAvarageRate);
+	std::sort(std::begin(arrayOfStudents), std::end(arrayOfStudents), sortByAvarageRate);
 
-    for (int i = 0; i < std::size(arrayOfStudents); i++)
-    {
-        printStudent(arrayOfStudents[i]);
-    }
+	for (int i = 0; i < std::size(arrayOfStudents); i++)
+	{
+		printStudent(arrayOfStudents[i]);
+	}
 
-    Student* ptrToBestStudent = findBest(arrayOfStudents, amountOfStudents);
-    std::cout << std::endl <<"Best Student => ";
-    printStudent(*ptrToBestStudent);
+	Student* ptrToBestStudent = findBest(arrayOfStudents, amountOfStudents);
+	std::cout << std::endl << "Best Student => ";
+	printStudent(*ptrToBestStudent);
 
-    int passedTreshold = sumStudentsPassedTreshold( arrayOfStudents, amountOfStudents, rateTreshold);
-    std::cout << std::endl << "Number of students which pass over 75% avarage rate: " << passedTreshold << std::endl;
+	int passedTreshold = sumStudentsPassedTreshold(arrayOfStudents, amountOfStudents, rateTreshold);
+	std::cout << std::endl << "Number of students which pass over 75% avarage rate: " << passedTreshold << std::endl;
 
 }
 
-float avaregeRateOfStudent(Student studentToClaculate)
+float averegeRateOfStudent(const Student& studentToClaculate)
 {
-    float avarageRate = 0;
-    for (int i = 0; i < std::size(studentToClaculate.rates); i++)
-    {
-        avarageRate += studentToClaculate.rates[i];
-    }
+	float averageRate = 0;
+	for (int i = 0; i < std::size(studentToClaculate.rates); i++)
+	{
+		averageRate += studentToClaculate.rates[i];
+	}
 
-    avarageRate /= static_cast<float>(std::size(studentToClaculate.rates));
+	averageRate /= static_cast<float>(std::size(studentToClaculate.rates));
 
-    return avarageRate;
+	return averageRate;
 }
 
-bool sortByAvarageRate(Student firstStudent, Student secondStudent) 
+bool sortByAvarageRate(const Student& firstStudent, const Student& secondStudent)
 {
-    float avarageRateFirstStudent = 0;
-    float avarageRateSecondStudent = 0;
+	float avarageRateFirstStudent = 0;
+	float avarageRateSecondStudent = 0;
 
-    for (int i = 0; i < std::size(firstStudent.rates); i++)
-    {
-        avarageRateFirstStudent += firstStudent.rates[i];
-    }
+	for (int i = 0; i < std::size(firstStudent.rates); i++)
+	{
+		avarageRateFirstStudent += firstStudent.rates[i];
+	}
 
-    avarageRateFirstStudent /= static_cast<float>(std::size(firstStudent.rates));
+	avarageRateFirstStudent /= static_cast<float>(std::size(firstStudent.rates));
```

Comment:
you can use it:
`float averegeRateOfStudent(const Student& studentToClaculate);`